src/eva/vision/data/transforms/common/resize_and_crop.py

```diff
@@ -35,7 +36,13 @@ def _build_transforms(self) -> Sequence[Callable]:
             torch_transforms.ToImage(),
             torch_transforms.Resize(size=self._size),
             torch_transforms.CenterCrop(size=self._size),
-            torch_transforms.ToDtype(torch.float32, scale=True),
+            torch_transforms.ToDtype(
+                {
+                    tv_tensors.Image: torch.float32,
```

Comment:
Is this compatible with the existing dataset classes which don't return `tv_tensors.Image` and `tv_tensors.Mask`?

Reply:
yes! its only if it finds these types of tensors will apply the transforms

Reply:
I see, but then for other datasets we don't transform to `torch.float32` anymore, which we previously did right?

Reply:
good question, but no, they will be still transformed to `torch.float32` and scaled to (0, 1); basically here torchvision acts smart: because we sent the images as `uint8`, it understands that they are images so it treats them as `tv_tensors.Image` and applies the transforms 

According to the docs, its basically `torch_transforms.ToDtype({tv_tensors.Image: torch.float32,torch.float32}) == torch_transforms.ToDtype(torch.float32)`, because if a torch.dtype is passed, e.g. `torch.float32`, only images and videos will be converted to that dtype